### PR TITLE
[Bug][Channel] The hostname resolver does not check for the channel status

### LINKED ISSUE #13677: [Bug][Channel] The hostname resolver does not check for the channel status
**Sylius version affected**: 1.*

**Description**  
The HostnameBasedRequestResolver searches the channel using the findOneByHostname method of the Channel Repository, but this method or the resolver itself does not check for the channel status (enabled/disabled). So a disabled channel can be navigated.

**Possible Solution**  
Add check for status in the repository (new method findOneActiveByHostname) or in the resolver.

### PR DESCRIPTION
**Pull request by lruozzi9**
| Q               | A
| --------------- | -----
| Branch?         | 1.10
| Bug fix?        |yes
| New feature?    | no
| BC breaks?      | no
| Deprecations?   | yes
| Related tickets | fixes #13677 
| License         | MIT

### CODE REVIEW

**Comment by Roshyo** [comment]
Hi @lruozzi9 . Could it be possible to come up with a testing scenario on that please ?

**Comment by lruozzi9** [comment]
Done!

**Review by Roshyo — APPROVED** [review]

**Review by Ferror — APPROVED** [review]

**Review by coldic3 — APPROVED** [review]

**Review by Zales0123 — COMMENTED** [review]
Also, it should target 1.11 as we no longer support 1.10 branch 🖖 Thanks for your contribution!

**Comment by lruozzi9** [comment]
> Also, it should target 1.11 as we no longer support 1.10 branch 🖖 Thanks for your contribution!

It was when I opened the PR 🤣.

File: features/channel/browsing_shop_on_right_channel.feature [hunk 1]
```diff
@@ -0,0 +1,15 @@
...
+
+    Background:
+        Given the store operates on a channel named "Default" in "USD" currency and with hostname "127.0.0.1"
+        And the store also operates on a channel named "Alternative" in "USD" currency and with hostname "127.0.0.1"
+        And the channel "Default" is disabled
+
+    @ui
+    Scenario: Browsing shop in active channel
+        When I visit the homepage
+        Then I should be on channel "Alternative"
```
[the remark this answers is not in the document]
Comment (lruozzi9) [reply features/channel/browsing_shop_on_right_channel.feature:15]: Right!

**Comment by GSadee** [comment]
Closing in favour of https://github.com/Sylius/Sylius/pull/14155

**Pull request closed without merging**